app/parse.py

```diff
@@ -14,9 +36,98 @@ class Product:
     rating: int
     num_of_reviews: int
 
+    @classmethod
+    def product_fields(cls) -> list[str]:
+        return [field.name for field in fields(cls)]
+
+
+class Scraper:
+    def __init__(self) -> None:
+        self.driver = ChromeDriver()
+
+    @staticmethod
+    def parse_single_product(product_soup: BeautifulSoup) -> Product:
+        return Product(
+            title=product_soup.select_one(".title")["title"],
+            description=product_soup.select_one(
+                ".description"
+            ).text.replace("\xa0", " "),
+            price=float(
+                product_soup.select_one(".price").text.replace("$", "")
+            ),
+            rating=int(len(product_soup.find(
+                "div", class_="ratings"
+            ).find_all("span", class_="ws-icon"))),
+            num_of_reviews=int(
+                product_soup.select_one(".review-count").text.split()[0]
+            )
+        )
+
+    def get_page_all_products(self, page_soup: BeautifulSoup) -> [Product]:
+        return [self.parse_single_product(product_soup)
+                for product_soup in page_soup.select(".card-body")]
+
+    def parse_all_products_on_page(self, url: str) -> [Product]:
+        self.driver.get(url)
+
+        try:
+            accept_cookies_button = WebDriverWait(self.driver, 2).until(
+                EC.element_to_be_clickable((By.CLASS_NAME, "acceptCookies"))
+            )
+            accept_cookies_button.click()
+            print("Cookies was accepted")
+        except TimeoutException:
+            print("No pop-up window accepting cookies")
+        except NoSuchElementException:
+            print("No item accepting cookies")
```

Comment:
what if cookies weren't accepted? you still proceed with scrapping? Then what is the point of this try except block?